### DISCUSSION TITLE
How do I connect to other nodes within the intranet

### DISCUSSION THREAD

**Discussion by DerekYeung** [OffchainLabs/prysm#11493] [Q&A]
I have several machines on the same LAN all running beacon nodes, is there any way to connect these nodes together?

**Comment by prestonvanloon** [comment]
You can use the `--peer` flag multiple times to connect nodes. Your peer address is printed at start up

**Comment by DerekYeung** [comment]
`--peer nodeAddress`?

**Comment by DerekYeung** [comment]
btw can I use one consensus client to serve multiple execution clients? @prestonvanloon thanks

**Comment by prestonvanloon** [comment]
> `--peer nodeAddress`?


Look for a message like this
```
[2022-09-23 12:10:10]  INFO p2p: Node started p2p server multiAddr=/ip4/192.168.1.25/tcp/13000/p2p/16Uiu2HAmLBZgZ3tGugCDVmYvFtVNFfZSAx3itNTNbPjxxcG8T5Q9
```

Then in your other client add the peer flag like this
```
--peer=/ip4/192.168.1.25/tcp/13000/p2p/16Uiu2HAmLBZgZ3tGugCDVmYvFtVNFfZSAx3itNTNbPjxxcG8T5Q9
```

You can specify the peer flag multiple times for each peer you want to add manually.

**Comment by prestonvanloon** [comment]
> btw can I use one consensus client to serve multiple execution clients? @prestonvanloon thanks

Not at this time. It's a bit more complicated than it sounds

**Comment by DerekYeung** [comment]
oh I see, I guess the description of this documentation confuses me, it says "Connect with this peer", I didn't realize "this" represented the target address lol

**Comment by DerekYeung** [comment]
> > btw can I use one consensus client to serve multiple execution clients? @prestonvanloon thanks
> 
> Not at this time. It's a bit more complicated than it sounds

got it, thanks!

**Comment by prestonvanloon** [comment]
> oh I see, I guess the description of this documentation confuses me, it says "Connect with this peer", I didn't realize "this" represented the target address lol

Is there somewhere we can improve that documentation? 

cc @mick-prysm (docs expert)

**Comment by DerekYeung** [comment]
--peer value            Connect with this peer. This flag may be used multiple times.  (accepts multiple inputs)

I perfer change to "Connect with target peer"
"this" made me mistake it for itself, personal feeling

**Comment by DerekYeung** [comment]
opps I have a question my prysm client is running via docker, so the address in the logs is `/ip4/172.18.0.3/tcp/13001/p2p/`, can I change 172 to my local ip? will that work or I need set something up?

**Comment by prestonvanloon** [comment]
Try setting `--p2p-host-ip` to the appropriate value. This is the IP that the node will advertise, and I think will affect the log message you see to have the correct IP.

**Comment by DerekYeung** [comment]
I currently set p2p-host-ip to my real ip and map it in the route

**Comment by DerekYeung** [comment]
let me try the local ip first, if it doesnt work I think the public ip should work right

**Comment by prestonvanloon** [comment]
Yeah, you can just replace the public IP with the private one. That ought to work.

**Reply by DerekYeung** [reply]
I just tried, so the public ip works, but the private one not, I think it could bc docker

**Reply by prestonvanloon** [reply]
Could be docker. The node will communicate all of its possible connection strings. So you may have discovered the node via public IP, but you might be talking to it on the local intranet. I'm not sure about your specific setup.

**Reply by DerekYeung** [reply]
ye it displayed the public ip + docker ip but not my private ip
this is my docker setup
```
    command: [
      "--rpc-host=0.0.0.0",
      "--grpc-gateway-host=0.0.0.0",
      "--grpc-gateway-port=3500",
      "--monitoring-host=0.0.0.0",
      "--p2p-tcp-port=13004",
      "--p2p-udp-port=12004",
      "--accept-terms-of-use",
      "--execution-endpoint=/eth/master/geth.ipc",
      "--datadir=/eth/cs/master",
      "--peer=/ip4/MyPublicIp/tcp/13001/p2p/16Uiu2HAm5Qxi9vwHz4znkButRcxZ14p4LqjJdpucH9nfmLxhwKTC"
    ]
```

**Reply by DerekYeung** [reply]
and the p2p address in the p2p endpoint show me
/ip4/MyPublicIp/tcp/13001/p2p/16Uiu2HAm5Qxi9vwHz4znkButRcxZ14p4LqjJdpucH9nfmLxhwKTC,/ip4/172.18.0.3/tcp/13001/p2p/16Uiu2HAm5Qxi9vwHz4znkButRcxZ14p4LqjJdpucH9nfmLxhwKTC

**Comment by DerekYeung** [comment]
@prestonvanloon is there anyway to check if its working?

**Comment by prestonvanloon** [comment]
Check http://localhost:8080/p2p to see your peers. If it's not on localhost, you may need to set `--monitoring-host=0.0.0.0`
